octodns/provider/mythicbeasts.py

```diff
@@ -0,0 +1,420 @@
+        values = []
+
+        for raw_value in \
+                [raw_values['value'] for raw_values in data['raw_values']]:
+
+            match = re.match(
+                '^([0-9]+)\\s+([0-9]+)\\s+([0-9]+)\\s+(\\S+)$',
+                raw_value,
+                re.IGNORECASE)
+
+            assert match is not None, 'Unable to parse SRV data'
+
+            target = match.group(4)
+            if not target.endswith('.'):
+                target = '{}.{}'.format(target, data['zone'])
+
+            values.append({
+                'priority': match.group(1),
+                'weight': match.group(2),
+                'port': match.group(3),
+                'target': target,
+            })
+
+        return {
+            'type': _type,
+            'values': values,
+            'ttl': ttl,
+        }
+
+    @staticmethod
+    def _data_for_SSHFP(_type, data):
+        ttl = max([raw_values['ttl'] for raw_values in data['raw_values']])
+        values = []
+
+        for raw_value in \
+                [raw_values['value'] for raw_values in data['raw_values']]:
+            match = re.match(
+                '^([0-9]+)\\s+([0-9]+)\\s+(\\S+)$',
+                raw_value,
+                re.IGNORECASE)
+
+            assert match is not None, 'Unable to parse SSHFP data'
+
+            values.append({
+                'algorithm': match.group(1),
+                'fingerprint_type': match.group(2),
+                'fingerprint': match.group(3),
+            })
+
+        return {
+            'type': _type,
+            'values': values,
+            'ttl': ttl,
+        }
+
+    @staticmethod
+    def _data_for_CAA(_type, data):
+        ttl = data['raw_values'][0]['ttl']
+        raw_value = data['raw_values'][0]['value']
+
+        match = re.match(
+            '^([0-9]+)\\s+(issue|issuewild|iodef)\\s+(\\S+)$',
+            raw_value,
+            re.IGNORECASE)
+
+        assert match is not None, 'Unable to parse CAA data'
+
+        value = {
+            'flags': match.group(1),
+            'tag': match.group(2),
+            'value': match.group(3),
+        }
+
+        return MythicBeastsProvider._data_for_single(
+            'CAA',
+            {'raw_values': [{'value': value, 'ttl': ttl}]})
+
+    _data_for_NS = _data_for_multiple
+    _data_for_TXT = _data_for_multiple
+    _data_for_A = _data_for_multiple
+    _data_for_AAAA = _data_for_multiple
+
+    def populate(self, zone, target=False, lenient=False):
+        self.log.debug('populate: name=%s, target=%s, lenient=%s', zone.name,
+                       target, lenient)
+
+        resp = self.records(zone.name)
+
+        before = len(zone.records)
+        exists = False
+        data = dict()
+
+        exists = True
+        for line in resp.content.splitlines():
+            match = re.match(
+                '^(\\S+)\\s+(\\d+)\\s+(\\S+)\\s+(.*)$',
+                line,
+                re.IGNORECASE)
```

Comment:
It is pretty hard to decipher which group is what here. Probably worth using named groups. `r'(?P<name>\S+...`
& `name = match.group('name')` sort of thing.  https://docs.python.org/2/library/re.html#re.MatchObject.group

Reply:
Sure will do.